### DISCUSSION TITLE
reference/reverse-proxy

### DISCUSSION THREAD

**Discussion by giscus** [Mereithhh/vanblog-comment#14] [Announcements]
# reference/reverse-proxy

注意 VanBlog 内置了 caddy，可以全自动申请 https 证书，如没有其他服务需要共存，是不建议再加一层反代的。反代时只需要反代映射的 HTTP 端口，由于 VanBlog 是一个整体，无需考虑内部的 Caddy。你需要： 1. 按需修改默认的 80 端口号 1. 关闭 https 自动重定向 (默认是关闭的) 反代方式 nginx-pro...

https://vanblog.mereith.com/reference/reverse-proxy.html

**Comment by king-bcolor** [comment]
**如果用docker容器安装nginx，需要在yml文件里加上 `network_mode: "host"`，容器默认与宿主机网络隔离**

**Reply by king-bcolor** [reply]
[基于docker的nginx反代来部署vanblog](http://kaifariji.top/post/6)

**Comment by yeliangzai** [comment]
一直按照步骤部署的，404报错

**Reply by yeliangzai** [reply]
不知道是漏了哪一步
·· <html>
<head><title>404 Not Found</title></head>
<body>
<center><h1>404 Not Found</h1></center>
<hr><center>nginx</center>
</body>
本地也无法进行访问

**Comment by xianjun-li** [comment]
我的用来做反代的环境也是使用另一份compose.yml文件启动的，反代服务器也是caddy,请问应该如何配置呢？

**Reply by Mereithhh** [reply]
1. 按需修改默认的 80 端口号 2. 关闭 https 自动重定向  3. 用你另一个 caddy 反代 http 端口
